Sources/Web3Core/Contract/ContractProtocol.swift

```diff
@@ -210,8 +210,8 @@ extension ContractProtocol {
 
     func decodeInputData(_ data: Data) -> [String: Any]? {
         guard data.count >= 4 else { return nil }
-        let methodId = data[0..<4].toHexString()
-        let data = data[4...]
+        let methodId = data[data.indices.startIndex ..< data.indices.startIndex + 4].toHexString()
+        let data = data[(data.indices.startIndex + 4)...]
```

Comment:
That needs to be covered with additional test cases passing a data slice instead of the original data object.

Reply:
It applies to all functions changed below.